geoips/create_plugin_registries.py

```diff
@@ -439,15 +481,25 @@ def add_module_plugin(package, relpath, plugins):
     abspath = resources.files(package) / relpath
     spec = util.spec_from_file_location(module_name, abspath)
     try:
+        from inspect import signature
+
         module = util.module_from_spec(spec)
         spec.loader.exec_module(module)
         interface_name = module.interface
         if interface_name not in plugins.keys():
             plugins[interface_name] = {}
         name = module.name
-        del module
         check_plugin_exists(package, plugins, interface_name, name)
-        plugins[interface_name][name] = {"relpath": relpath}
+        plugins[interface_name][name] = {
+            "docstring": module.__doc__,
+            "family": module.family,
+            "interface": module.interface,
+            "package": package,
+            "plugin_type": "module_based",
+            "signature": str(signature(module.call)),
+            "relpath": relpath,
+        }
+        del module
```

Comment:
Here we are acquiring all of the attributes required on a module based plugin, and ensuring everything ends up in the registry?  Again, maybe leave more comments.

Reply:
I can leave more comments. Yes, it is here where we are acquiring all of the required attributes for a module_plugin.

Reply:
https://github.com/NRLMMD-GEOIPS/geoips/pull/392/commits/6754848389fd6b02250be244157016c28f81a133